File: 151001/Novik/RESTLab1/REST/Mapper/PostMapper.cs
using AutoMapper;
using REST.Models.DTOs.RequestIn;
using REST.Models.DTOs.Responses;
using REST.Models.Entity;

namespace REST.Mapper;

public class PostMapper : Profile
{

    public PostMapper()
    {
        CreateMap<Post, PostResponseTo>();
        CreateMap<PostRequestTo, Post>();
    }
    
}

File: 151002/Luschickij/lab2/DC_lab2/src/main/java/com/luschickij/DC_lab/service/mapper/PostMapper.java
package com.luschickij.DC_lab.service.mapper;

import com.luschickij.DC_lab.model.entity.Post;
import com.luschickij.DC_lab.model.request.PostRequestTo;
import com.luschickij.DC_lab.model.response.PostResponseTo;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import java.util.List;

@Mapper(componentModel = "spring", uses = CustomMapper.class)
public interface PostMapper {
    @Mapping(source = "news.id", target = "newsId")
    PostResponseTo getResponse(Post post);
    @Mapping(source = "news.id", target = "newsId")
    List<PostResponseTo> getListResponse(Iterable<Post> posts);
    @Mapping(source = "newsId", target = "news", qualifiedByName = "newsRefFromNewsId")
    Post getPost(PostRequestTo postRequestTo);
}
